src/data-types/mailstream_helper.c

```diff
@@ -62,16 +62,21 @@ char * mailstream_read_line(mailstream * stream, MMAPString * line)
   return mailstream_read_line_append(stream, line);
 }
 
-static char * mailstream_read_len_append(mailstream * stream,
-					 MMAPString * line,
-					 size_t i)
+static char * mailstream_read_len_append_from_buffer(mailstream * stream,
```

Comment:
The mailstream_read_len_append function always reads from buffer. I try reflect this fact in their name.
I undo this change.